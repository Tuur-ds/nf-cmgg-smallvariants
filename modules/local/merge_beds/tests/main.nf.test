nextflow_process {

    name "Test Process MERGE_BEDS"
    script "../main.nf"
    process "MERGE_BEDS"

    tag "modules"
    tag "modules_local"
    tag "merge_beds"

    test("homo_sapiens - bed") {

        config "./nextflow.config"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.bed, checkIfExists:true)
                ]
                input[1] = [
                    [ id:'fai' ],
                    file(params.fai, checkIfExists:true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("homo_sapiens - bed - stub") {

        options "-stub"
        config "./nextflow.config"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.bed, checkIfExists:true)
                ]
                input[1] = [
                    [ id:'fai' ],
                    file(params.fai, checkIfExists:true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}