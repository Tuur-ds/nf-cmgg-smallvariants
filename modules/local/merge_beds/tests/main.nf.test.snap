{
    "homo_sapiens - bed": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,fa245abf8add7a80650566a1de67ec04"
                    ]
                ],
                "1": [
                    "versions.yml:md5,273f887b675fb5feb6073a5313a191a9"
                ],
                "bed": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,fa245abf8add7a80650566a1de67ec04"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,273f887b675fb5feb6073a5313a191a9"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-14T09:49:08.980488978"
    },
    "homo_sapiens - bed - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,273f887b675fb5feb6073a5313a191a9"
                ],
                "bed": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,273f887b675fb5feb6073a5313a191a9"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-14T09:49:17.089675797"
    }
}